### PR TITLE
Sa/add detail screen

### PR DESCRIPTION
**Pull request by SultanArshad**

### CODE REVIEW

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/util/NetworkConnectionInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.io.IOException
+
+class NetworkConnectionInterceptor
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/util/NetworkConnectionInterceptor.kt:11]: do we need this for this PR? this PR is about details screen right? 🤔

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import kotlinx.coroutines.*
+import org.koin.core.component.*
+import org.koin.core.parameter.parametersOf
+import org.koin.core.qualifier.named
+import org.koin.core.scope.Scope
+import kotlin.coroutines.CoroutineContext
+
+class DetailViewController(bundle: Bundle) : Controller(bundle), KoinComponent,
+  CoroutineScope {
+  private val pokemon: Pokemon = bundle.getParcelable(POKEMON)!!
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt:31]: avoid ‼️, instead be explicit about your assetion 
```kotlin
private val pokemon: Pokemon = requireNotNull(bundle.getParcelable(POKEMON)) { 
  "Details view controller attached without a pokemon" 
}
```

File: app-kotlin/src/main/res/layout/detail_view_controller.xml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <include layout="@layout/error_view" />
+
+    <FrameLayout
+        android:id="@+id/detail_view"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:visibility="gone">
```
Comment (xxfast) [inline app-kotlin/src/main/res/layout/detail_view_controller.xml:13]: do we need this nested framelayout?

File: app-kotlin/src/main/res/layout/error_view.xml [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
-    android:layout_height="100dp"
-    android:layout_gravity="center"
-    android:visibility="gone"
-    app:contentDescription="error image"
-    app:srcCompat="@drawable/ic_error_round" />
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
 
+    <ProgressBar
+        android:id="@+id/home_progressBar"
```
Comment (xxfast) [inline app-kotlin/src/main/res/layout/error_view.xml:8]: this is reused by both `home` and `details` right? 🤔 calling this home makes less sense now

**Review by xxfast — CHANGES_REQUESTED** [review]
Nice work 👍

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/util/NetworkConnectionInterceptor.kt (hunk 1 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/util/NetworkConnectionInterceptor.kt:11]: fixed

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt (hunk 2 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt:31]: yeah sure updating this to be explicit thanks.

File: app-kotlin/src/main/res/layout/detail_view_controller.xml (hunk 3 above)
Comment (SultanArshad) [reply app-kotlin/src/main/res/layout/detail_view_controller.xml:13]: Yes, we do need this to hide the view when the network called failed and an error sign appears.

File: app-kotlin/src/main/res/layout/error_view.xml (hunk 4 above)
Comment (SultanArshad) [reply app-kotlin/src/main/res/layout/error_view.xml:8]: Good point I had changed it to progressBar

File: app-kotlin/src/main/res/layout/error_view.xml [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
-    app:srcCompat="@drawable/ic_error_round" />
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
 
+    <ProgressBar
+        android:id="@+id/progressBar"
+        android:layout_width="20dp"
+        android:layout_height="20dp"
+        android:layout_gravity="center"
+        android:visibility="gone" />
```
Comment (xxfast) [inline app-kotlin/src/main/res/layout/error_view.xml:12]: progress bar in a error view? 🤔 i feel `error_view` is a misnomer

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import org.koin.core.parameter.parametersOf
+import org.koin.core.qualifier.named
+import org.koin.core.scope.Scope
+import kotlin.coroutines.CoroutineContext
+
+class DetailViewController(bundle: Bundle) : Controller(bundle), KoinComponent,
+  CoroutineScope {
+  private val pokemon: Pokemon = requireNotNull(bundle.getParcelable(POKEMON)) {
+    "Details view controller attached without a pokemon"
+  }
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt:33]: group your private members together

**Review by xxfast — CHANGES_REQUESTED** [review]
Nice work 👍

File: app-kotlin/src/main/res/layout/error_view.xml (hunk 5 above)
Comment (SultanArshad) [reply app-kotlin/src/main/res/layout/error_view.xml:12]: yes this is what i exactly think in recent pr to do thanks for mentioning i am fixing this in amend commit

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    val progressBar: ProgressBar = view.findViewById(R.id.progressBar)
+    val errorImage: ImageView = view.findViewById(R.id.image_error)
+    val name: TextView = view.findViewById(R.id.name)
+    val attack: TextView = view.findViewById(R.id.attack_value)
+    val spAttack: TextView = view.findViewById(R.id.sp_attack_value)
+    val defense: TextView = view.findViewById(R.id.defense_value)
+    val spDefense: TextView = view.findViewById(R.id.sp_defense_value)
+    val speed: TextView = view.findViewById(R.id.speed_value)
+    val hp: TextView = view.findViewById(R.id.hp_value)
+    val image: ImageView = view.findViewById(R.id.image)
```
[the remark this answers is not in the document]
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/DetailViewController.kt:58]: Thanks for updating i will follow this style from now thanks

File: app-kotlin/src/main/res/layout/detail_view_controller.xml (hunk 3 above)
Comment (xxfast) [reply app-kotlin/src/main/res/layout/detail_view_controller.xml:13]: I think you forgot to remove this second frame layout 🤔

**Review by xxfast — CHANGES_REQUESTED** [review]
Almost there 👍

**Review by xxfast — APPROVED** [review]
Nice work 👍

**Pull request merged**